url/src/lib.rs

```diff
@@ -1709,6 +1709,33 @@ impl Url {
         let old_after_path_pos = to_u32(self.serialization.len()).unwrap();
         let cannot_be_a_base = self.cannot_be_a_base();
         let scheme_type = SchemeType::from(self.scheme());
+        let mut path_empty = false;
+        let mut has_host = if let Some(index) = self.serialization.find(":/") {
```

Comment:
I'm a bit worried that this would also match URL's such as `scheme:hello:/fakehost`.
Can you instead find `:` and then check if the next character is `/` ?